packages/smart-contracts/src/contracts/ERC20EscrowToPayV1.sol

```diff
@@ -0,0 +1,385 @@
+/// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.7;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/TokenTimelock.sol";
+import "./interfaces/ERC20FeeProxy.sol";
+
+/**
+ * @title   ERC20EscrowToPayV1.
+ * @author  Ivo Garofalo.
+ * @notice  Request Invoice with Escrow.
+ */
+contract ERC20EscrowToPayV1 {
+    IERC20FeeProxy paymentProxy;
+    address payable owner;
+
+    struct Invoice {
+        IERC20 paymentToken; 
+        uint256 amount;
+        address payee;
+        address payer;
+        uint feeAmount;
+        address feeAddress;
+        uint256 claimDate;
+    }
+
+    struct Dispute {
```

Comment:
```suggestion
    struct FrozenPayment {
```
Or maybe another name ? For example "FundsLocks" or "FrozenEscrow" etc.

But Dispute refers to the general "process" of dispute, which may include funds locks or other better suited options (for example: agree on 50-50 etc.)

Reply:
I see your point. 

What about "FrozenRequest"? 

Since the struct `Invoice` will be renamed to `Request`, it would seem logical to use `FrozenRequest` instead of `FrozenPayment`?

Reply:
```suggestion
    struct FrozenRequest {
```